aspnetcore/log-mon/metrics/metrics.md

```diff
@@ -175,6 +175,53 @@ Alternatively, enter counter category such as `kestrel` in the **Expression** in
 
 ![dashboard-screenshot2](~/log-mon/metrics/metrics/static/dashboard-screenshot.png)
 
+## Test metrics in ASP.NET Core apps
+
+It's possible to test metrics in ASP.NET Core apps. One way to do that is collect and assert metrics values in [ASP.NET Core integration tests](xref:test/integration-tests) using <xref:Microsoft.Extensions.Telemetry.Testing.Metering.MetricCollector%601>.
+
+```cs
+public class BasicTests : IClassFixture<WebApplicationFactory<Program>>
+{
+    private readonly WebApplicationFactory<Program> _factory;
+    public BasicTests(WebApplicationFactory<Program> factory) => _factory = factory;
+
+    [Fact]
+    public async Task Get_RequestCounterIncreased()
+    {
+        // Arrange
+        var client = _factory.CreateClient();
+        var meterFactory = _factory.Services.GetRequiredService<IMeterFactory>();
+        var collector = new MetricCollector<double>(meterFactory,
+            "Microsoft.AspNetCore.Hosting", "http.server.request.duration");
+
+        // Act
+        var response = await client.GetAsync("/");
+
+        // Assert
+        Assert.Equal("Hello World!", await response.Content.ReadAsStringAsync());
+
+        await collector.WaitForMeasurementsAsync(minCount: 1);
```

Comment:
If the code under test does not produce the intended metric, is this just going to hang indefinitely until some higher level test timeout occurs?

Reply:
Yes

Reply:
I added WaitAsync - https://github.com/dotnet/AspNetCore.Docs/pull/30091